documentation/projects/proposals/popularity_optimizations/20230420-implementation_plan_popularity_optimizations.md

```diff
@@ -0,0 +1,486 @@
+# Implementation Plan: Decoupling Popularity Calculations from the Data Refresh
+
+<!-- See the implementation plan guide for more information: https://github.com/WordPress/openverse/tree/19791f51c063d0979112f4b9f4eeace04c8cf5ff/docs/projects#implementation-plans-status-in-rfc -->
+<!-- This template is exhaustive and may include sections which aren't relevant to your project. Feel free to remove any sections which would not be useful to have. -->
+
+## Reviewers
+
+<!-- Choose two people at your discretion who make sense to review this based on their existing expertise. Check in to make sure folks aren't currently reviewing more than one other proposal or RFC. -->
+
+- [ ] @AetherUnbound
+- [ ] @obulat
+
+## Project links
+
+<!-- Enumerate any references to other documents/pages, including milestones and other plans -->
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/433)
+- [Project Proposal](https://docs.openverse.org/projects/proposals/popularity_optimizations/20230406-project_proposal_popularity_optimizations.md)
+
+## Overview
+
+<!-- A brief one or two sentence overview of the implementation being described. -->
+
+This document describes a method for separating the popularity calculation steps
+from the data refresh process, allowing the data refresh to be re-enabled
+independently of popularity calculations.
+
+## Background
+
+To improve search relevancy, we boost records in our search results relative to
+a normalized popularity score. Because the process of calculating these scores
+and pushing them into the API DB (referred to as "data refresh") is complex, it
+may be beneficial to briefly outline the tables and views currently used.
+
+For simplicity, I will use `image` as an example, but note that equivalent
+structures exist for `audio` and will exist for any media types added in the
+future. Descriptions are not exhaustive; some details that are not relevant have
+been omitted for brevity.
+
+#### `image`
+
+The 'main' image table. Rows in this table may contain **raw** popularity data
+in the `meta_data` column, such as the number of times this image has been
+viewed or downloaded.
```

Comment:
Maybe worth disambiguating here (or elsewhere) that this popularity data is sourced in the provider as opposed to Openverse-specific popularity data we could derive in the future since adding analytics to the frontend (an explicit goal of adding those).

The project proposal also does not disambiguate this.

Reply:
It was a tough balance to decide how much of the existing structure to document while aiming for brevity, and a lot is left out here if it is not affected or changed by the proposal, including almost everything about the scores are actually computed. We should add much more extensive documentation of the calculation as part of this project, though.

I think since I used an example of raw popularity data here it's definitely worth disambiguating that I mean `the number of times this image has been viewed or downloaded at the source provider`, _not_ views on Openverse. Good call.